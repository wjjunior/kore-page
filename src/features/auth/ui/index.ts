// AuthButton component was removed and integrated into Header.vue
