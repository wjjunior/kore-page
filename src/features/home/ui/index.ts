export { default as HomeBanner } from "./HomeBanner.vue";
