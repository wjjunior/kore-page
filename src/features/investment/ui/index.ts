export { default as InvestmentOverview } from "./InvestmentOverview.vue";
export { default as InvestmentDetails } from "./InvestmentDetails.vue";
export { default as InvestmentCards } from "./InvestmentCards.vue";
export { default as InvestmentTeam } from "./InvestmentTeam.vue";
export { default as InvestmentMarketingPlan } from "./InvestmentMarketingPlan.vue";
export { default as InvestmentFaq } from "./InvestmentFaq.vue";
