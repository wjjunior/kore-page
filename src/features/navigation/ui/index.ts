// NavigationLink component was removed and integrated into Header.vue
