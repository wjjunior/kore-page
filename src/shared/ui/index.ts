export { default as MobileMenuIcon } from "./MobileMenuIcon.vue";
