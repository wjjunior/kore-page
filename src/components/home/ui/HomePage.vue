<template>
  <div>
    <Header active-nav-item="invest" />
    <HomeBanner />
    <div
      class="min-h-screen bg-gradient-to-br from-blue-50 to-indigo-100 flex items-center justify-center"
    >
      <div class="bg-white rounded-lg shadow-xl p-8 max-w-md w-full mx-4">
        <h1 class="text-3xl font-bold text-gray-800 mb-4 text-center">
          Homepage
        </h1>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { onMounted } from "vue";
import { Header } from "@/widgets/header";
import HomeBanner from "@/features/home/ui/HomeBanner.vue";
import { useInvestmentData } from "@/features/home";

defineOptions({
  name: "HomePage",
});

const { loadData } = useInvestmentData();

onMounted(async () => {
  await loadData();
});
</script>
