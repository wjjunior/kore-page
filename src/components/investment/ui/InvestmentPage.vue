<template>
  <div>
    <Header active-nav-item="invest" />
    <InvestmentOverview />
    <div
      class="min-h-screen bg-gradient-to-br from-blue-50 to-indigo-100 flex items-center justify-center"
    >
      <div class="bg-white rounded-lg shadow-xl p-8 max-w-md w-full mx-4">
        <h1 class="text-3xl font-bold text-gray-800 mb-4 text-center">
          Investment Page
        </h1>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { onMounted } from "vue";
import { Header } from "@/widgets/header";
import InvestmentOverview from "@/features/investment/ui/InvestmentOverview.vue";
import { useInvestmentData } from "@/features/investment";

defineOptions({
  name: "InvestmentPage",
});

const { loadData } = useInvestmentData();

onMounted(async () => {
  await loadData();
});
</script>
