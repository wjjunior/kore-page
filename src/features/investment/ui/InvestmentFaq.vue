<template>
  <div>
    <h2
      class="font-hanken-grotesk font-bold text-[32px] leading-[100%] tracking-[1%] text-primary-200 mb-6"
    >
      FAQ
    </h2>

    <div v-if="loading" class="space-y-4">
      <div v-for="i in 4" :key="i" class="animate-pulse">
        <div class="h-12 bg-gray-200 rounded border border-primary-200"></div>
      </div>
    </div>

    <div v-else class="flex flex-col gap-2">
      <AccordionItem
        v-for="(item, index) in faqItems"
        :key="item.id"
        :title="item.question"
        :content="item.answer"
        :is-open="openItems[index]"
        @toggle="toggleAccordion(index)"
      />
    </div>
  </div>
</template>

<script setup lang="ts">
import { ref } from "vue";
import { AccordionItem } from "@/shared/ui";

interface FaqItem {
  id: string;
  question: string;
  answer: string;
}

interface InvestmentFaqProps {
  loading?: boolean;
  faqItems?: FaqItem[];
}

withDefaults(defineProps<InvestmentFaqProps>(), {
  loading: false,
  faqItems: () => [
    {
      id: "faq-1",
      question: "Lorem ipsum",
      answer:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
    },
    {
      id: "faq-2",
      question: "Lorem ipsum",
      answer:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
    },
    {
      id: "faq-3",
      question: "Lorem ipsum",
      answer:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
    },
    {
      id: "faq-4",
      question: "Lorem ipsum",
      answer:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
    },
  ],
});

const openItems = ref<Record<number, boolean>>({});

const toggleAccordion = (index: number) => {
  openItems.value[index] = !openItems.value[index];
};
</script>
