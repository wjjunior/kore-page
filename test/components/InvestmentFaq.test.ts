import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import InvestmentFaq from "@/features/investment/ui/InvestmentFaq.vue";
import { AccordionItem } from "@/shared/ui";
import { mockFaqItems } from "../mocks";

const mountInvestmentFaq = (props: any = { faqItems: mockFaqItems }) => {
  return mount(InvestmentFaq, {
    props,
    global: {
      components: {
        AccordionItem,
      },
    },
  });
};

describe("InvestmentFaq", () => {
  it("renders the FAQ title", () => {
    const wrapper = mountInvestmentFaq();
    const heading = wrapper.find("h2");
    expect(heading.exists()).toBe(true);
    expect(heading.text()).toBe("FAQ");
  });

  it("displays all AccordionItems when not loading", () => {
    const wrapper = mountInvestmentFaq({
      faqItems: mockFaqItems,
      loading: false,
    });
    const accordionItems = wrapper.findAllComponents(AccordionItem);
    expect(accordionItems).toHaveLength(mockFaqItems.length);
  });

  it("does not render AccordionItems when loading", () => {
    const wrapper = mountInvestmentFaq({
      faqItems: mockFaqItems,
      loading: true,
    });
    expect(wrapper.findAllComponents(AccordionItem)).toHaveLength(0);
  });

  it("displays loading skeletons when loading", () => {
    const wrapper = mountInvestmentFaq({
      faqItems: mockFaqItems,
      loading: true,
    });
    const skeletons = wrapper.findAll(".animate-pulse");
    expect(skeletons).toHaveLength(4);
  });

  it("renders AccordionItems with correct props", () => {
    const wrapper = mountInvestmentFaq({ faqItems: mockFaqItems });
    const accordionItems = wrapper.findAllComponents(AccordionItem);

    accordionItems.forEach((itemWrapper, index) => {
      expect(itemWrapper.props("title")).toBe(mockFaqItems[index]?.question);
      expect(itemWrapper.props("content")).toBe(mockFaqItems[index]?.answer);
    });
  });

  it("renders no AccordionItems if faqItems is empty", () => {
    const wrapper = mountInvestmentFaq({ faqItems: [] });
    expect(wrapper.findAllComponents(AccordionItem)).toHaveLength(0);
  });

  it("toggles accordion item open state", () => {
    const wrapper = mountInvestmentFaq({ faqItems: mockFaqItems });
    const firstItem = wrapper.findComponent(AccordionItem);

    expect(firstItem.props("isOpen")).toBeFalsy();

    firstItem.vm.$emit("toggle");

    expect(firstItem.exists()).toBe(true);
  });

  it("matches snapshot", () => {
    const wrapper = mountInvestmentFaq({ faqItems: mockFaqItems });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
