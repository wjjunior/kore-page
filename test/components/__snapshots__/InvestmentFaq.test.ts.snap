// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InvestmentFaq > matches snapshot 1`] = `
"<div>
  <h2 class="font-hanken-grotesk font-bold text-2xl md:text-3xl lg:text-[30px] xl:text-[32px] leading-[100%] tracking-[1%] text-primary-200 mb-6"> FAQ </h2>
  <div class="flex flex-col gap-2">
    <div class="border border-primary-200 rounded-lg bg-white p-3 md:p-4 cursor-pointer hover:bg-gray-50 transition-colors w-full">
      <div class="flex items-center justify-between"><span class="font-hanken-grotesk text-primary-200 text-sm md:text-base pr-2 flex-1">What is this investment?</span><svg class="svg-inline--fa fa-chevron-down text-primary-200 transition-transform w-3 h-3 md:w-[12.53px] md:h-[22.2px] flex-shrink-0" data-prefix="fas" data-icon="chevron-down" role="img" viewBox="0 0 448 512" aria-hidden="true">
          <path class="" fill="currentColor" d="M201.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 338.7 54.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"></path>
        </svg></div>
      <!--v-if-->
    </div>
    <div class="border border-primary-200 rounded-lg bg-white p-3 md:p-4 cursor-pointer hover:bg-gray-50 transition-colors w-full">
      <div class="flex items-center justify-between"><span class="font-hanken-grotesk text-primary-200 text-sm md:text-base pr-2 flex-1">How long is the term?</span><svg class="svg-inline--fa fa-chevron-down text-primary-200 transition-transform w-3 h-3 md:w-[12.53px] md:h-[22.2px] flex-shrink-0" data-prefix="fas" data-icon="chevron-down" role="img" viewBox="0 0 448 512" aria-hidden="true">
          <path class="" fill="currentColor" d="M201.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 338.7 54.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"></path>
        </svg></div>
      <!--v-if-->
    </div>
  </div>
</div>"
`;
